crates/task-impls/src/request.rs

```diff
@@ -58,6 +59,8 @@ pub struct NetworkRequestState<
     pub _phantom: PhantomData<fn(&Ver)>,
     /// The node's id
     pub id: u64,
+    /// Vector of shutdown flags for delayed requesters
+    pub shutdown_flags: Vec<Arc<AtomicI8>>,
```

Comment:
Can this just be 1 atomic bool that every single delayed requester has.  Then in `cancel_vid` we just check if it's true and return true.  That way we don't have to do all the logic with the vector